classes/core/PKPContainer.inc.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+import('lib.pkp.classes.core.PKPEventServiceProvider');
+
+use Illuminate\Container\Container;
+use Illuminate\Config\Repository;
+use Illuminate\Support\Facades\Facade;
+
+class PKPContainer extends Container {
+
+	/**
+	 * @return void
+	 * @brief Create own container instance, initialize bindings
+	 */
+	public function __construct() {
+		$this->registerBaseBindings();
+		$this->registerCoreContainerAliases();
+	}
+
+	/**
+	 * @return void
+	 * @brief Bind the current container and set it globally
+	 * let helpers, facades and services know to which container refer to
+	 */
+	protected function registerBaseBindings() {
+
+		static::setInstance($this);
+		$this->instance('app', $this);
+		$this->instance(Container::class, $this);
+
+		Facade::setFacadeApplication($this);
+		// Load main settings, this should be done before registering services, e.g., it's used by Database Service
+		$this->loadConfiguration();
+	}
+
+	/**
+	 * @return void
+	 * @brief Register used service providers within the container
+	 */
+	public function registerConfiguredProviders() {
+		$this->register(new PKPEventServiceProvider($this));
+		$this->register(new Illuminate\Database\DatabaseServiceProvider($this));
+		$this->register(new Illuminate\Bus\BusServiceProvider($this));
+		$this->register(new Illuminate\Queue\QueueServiceProvider($this));
+	}
+
+	/**
+	 * @param Illuminate\Support\ServiceProvider $provider
+	 * @return void
+	 * @brief Simplified service registration
+	 */
+	public function register($provider) {
+		$provider->register();
+		if (method_exists($provider, 'boot')) {
+			$provider->boot();
+		}
+	}
+
+	/**
+	 * @return void
+	 * @brief Bind aliases with contracts
+	 */
+	public function registerCoreContainerAliases() {
+		foreach([
+			'app'              => [self::class, Illuminate\Contracts\Container\Container::class, Psr\Container\ContainerInterface::class],
+	        'config'           => [Illuminate\Config\Repository::class, Illuminate\Contracts\Config\Repository::class],
+	        'db'               => [Illuminate\Database\DatabaseManager::class, Illuminate\Database\ConnectionResolverInterface::class],
+	        'db.connection'    => [Illuminate\Database\Connection::class, Illuminate\Database\ConnectionInterface::class],
+	        'events'           => [Illuminate\Events\Dispatcher::class, Illuminate\Contracts\Events\Dispatcher::class],
+	        'queue'            => [Illuminate\Queue\QueueManager::class, Illuminate\Contracts\Queue\Factory::class, Illuminate\Contracts\Queue\Monitor::class],
+	        'queue.connection' => [Illuminate\Contracts\Queue\Queue::class],
+	        'queue.failer'     => [Illuminate\Queue\Failed\FailedJobProviderInterface::class],
+        ] as $key => $aliases) {
```

Comment:
Looks like some spaces need to be converted to tabs here.

Reply:
Indeed